src/Cns.ts

```diff
@@ -167,32 +176,43 @@ export default class Cns extends EthereumNamingService {
     return await this.record(domain, 'ipfs.redirect_domain.value');
   }
 
-  private async getTtl(
-    contract: Contract,
-    methodname: string,
-    params: string[],
-  ): Promise<string> {
-    return await this.callMethod(contract, methodname, params);
-  };
-
   /** @internal */
   async record(domain: string, key: string): Promise<string> {
     const tokenId = this.namehash(domain);
-    const resolver = await this.resolver(domain);
-    const resolverContract = this.buildContract(resolverInterface, resolver);
-    const record = await this.getRecord(resolverContract, 'get', [
-      key,
-      tokenId,
-    ]);
-    return this.ensureRecordPresence(domain, key, record);
-  }
-
-  /** This is done to make testwriting easy */
-  private async getRecord(
-    contract: Contract,
-    methodname: string,
-    params: any[],
-  ): Promise<any> {
-    return await this.callMethod(contract, methodname, params);
+
+    const reader = await this.getReader();
+    const data = await reader.record(tokenId, key);
+    await this.verify(domain, data);
+
+    const { values } = data;
+    const value: string | null = values?.length ? values[0] : null;
+    return this.ensureRecordPresence(domain, key, value);
+  }
+
+  /** @internal */
+  protected async getResolver(tokenId: string): Promise<string> {
```

Comment:
This seem to be unused now.

Reply:
true, but this is abstract function of `EthereumNamingService`

Reply:
It is fine to move it only to `Ens` if it is possible. `EthereumNamingService` is not a public API

Reply:
I see its impossible now. Lets just rewrite this method using proxy reader if possible.

Reply:
the method is not needed in `Cns.ts`. it can be changed to:
```
protected async getResolver(tokenId: string): Promise<string> {
    throw new Error('Method not implemented.');
}
```